RadixWallet/Core/SharedModels/Assets/OnLedgerEntity.swift

```diff
@@ -178,17 +208,27 @@ extension OnLedgerEntity {
 		case dappDefinition = "dapp definition"
 	}
 
-	public struct Resource: Sendable, Hashable, Codable, Identifiable {
+	public struct Resource: Sendable, Hashable, Codable, Identifiable, CustomDebugStringConvertible {
 		public var id: ResourceAddress { resourceAddress }
 		public let resourceAddress: ResourceAddress
 		public let atLedgerState: AtLedgerState
-		public let divisibility: Int?
+		public let divisibility: UInt8?
```

Comment:
Negative divisibility makes no sense

Reply:
It does in theory, if you only support even hundreds, thousands etc of a coin, but we probably don't support that.